### ISSUE TITLE
Release version of .NET Core 2.1 is incompatible - "Specified framework not found [...] version '2.1.0-preview1-26216-03"

### ISSUE THREAD

**Issue by stefan-schweiger** [RSuter/NSwag#1359]
I upgraded my project to the freshly released .NET Core 2.1. But it seems that nswag is looking for a preview version (which I don't have installed) and fails because it doesn't recognize the release version as compatible.

Probably rebuilding with the release version would solve this issue.

```
> nswag run /runtime:NetCore21

It was not possible to find any compatible framework version
The specified framework 'Microsoft.NETCore.App', version '2.1.0-preview1-26216-03' was not found.
  - Check application dependencies and target a framework version installed at:
      C:\Program Files\dotnet\
  - Installing .NET Core prerequisites might help resolve this problem:
      http://go.microsoft.com/fwlink/?LinkID=798306&clcid=0x409
  - The .NET Core framework and SDK can be installed from:
      https://aka.ms/dotnet-download
  - The following versions are installed:
      2.0.0 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.0.5 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.0.6 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.0.7 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
      2.1.0 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
```

**Comment by RSuter** [comment]
Yes, it's currently compiled against a preview version of .NET Core 2.1

**Comment by stefan-schweiger** [comment]
Since I couldn't find a download anymore for the preview version, my workaround for now is to simply duplicate the `2.1` folder in `C:\Program Files\dotnet\shared\Microsoft.NETCore.App` and rename it to `2.1.0-preview1-26216-03`.

This will work just fine until a new version is released.

**Comment by RSuter** [comment]
I'm wondering where this `2.1.0-preview1-26216-03` is stored... in my projects I only specified netcoreapp2.1... is this compiled into the dll when compiling with a preview version?

**Comment by RSuter** [comment]
Let's see if the CI build works against the latest SDK version: https://ci.appveyor.com/project/rsuter/nswag-25x6o/build/3155

**Comment by stefan-schweiger** [comment]
@RSuter I also didn't expect this behavior, but it seems like it's compiled into the dll. I guess running 2.0 on 2.1 would be less of a problem, but version detection with preview releases is often a bit funky.

**Comment by RSuter** [comment]
CI is not working anymore whereas it worked with the 2.1 preview... oh well...

**Comment by LeonidEfremov** [comment]
I have similar issue on my projects and I fix it with `global.json`
```
{
  "sdk": {
   "version": "2.1.300"
  }
}
```

**Comment by kachalkov** [comment]
global.json did not work for us, how can we redirect version for nswag for now?

**Comment by RSuter** [comment]
Should be fixed in v11.17.13, please report any problems...

- Release CI build: https://ci.appveyor.com/project/rsuter/nswag/build/991
- AppVeyor problem: https://help.appveyor.com/discussions/problems/12762-automated-net-core-tests-fail-to-run-if-the-any-cpu-platform-is-explicitly-specified
- Fix: https://github.com/RSuter/NSwag/commit/57bc77e114a2af65edc5bacf0963245a0f56b274
